src/EditorFeatures/CSharpTest/CodeActions/ExtractLocalFunction/ExtractLocalFunctionTests.cs

```diff
@@ -0,0 +1,3188 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.ExtractLocalFunction;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings.ExtractMethod
+{
+    public class ExtractLocalFunctionTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new CSharpExtractLocalFunctionCodeRefactoringProvider();
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsExtractLocalFunction)]
+        public async Task TestPartialSelection_StaticOptionTrue()
+        {
+            await TestInRegularAndScriptAsync(
```

Comment:
do you have tests that validate where extract-local-function should *not* be allowed?  Specifically cases like this:

1. field intializer
2. property initializer
3. attribute initializer
4. this/base constructor call

?

Reply:
Just added these. 👍 thank you for the recommendation!